### PR TITLE
NFT-964: Fix swap impact projection not working when not using underlying

### PR DESCRIPTION
**Pull request by adamgobes**
https://user-images.githubusercontent.com/7279416/222211260-37292631-4b45-4659-ab7e-c1196a0c9c52.mov

### CODE REVIEW

File: components/SwapPageContent/SwapPageContent.tsx [hunk 1]
```diff
@@ -113,46 +114,65 @@ export function SwapPageContent() {
...
+      baseCurrency = underlyingUniswapToken;
+      quoteCurrency = paprUniswapToken;
+    } else if (paprOut && amounts.tradeType === TradeType.EXACT_INPUT) {
+      baseCurrency = underlyingUniswapToken;
+      quoteCurrency = paprUniswapToken;
+    } else if (paprOut && amounts.tradeType === TradeType.EXACT_OUTPUT) {
+      baseCurrency = paprUniswapToken;
+      quoteCurrency = underlyingUniswapToken;
+    } else {
+      throw new Error('Invalid trade type');
```
Comment (wilsoncusack) [inline components/SwapPageContent/SwapPageContent.tsx:173]: what would be this case?

File: components/SwapPageContent/SwapPageContent.tsx [hunk 2]
```diff
@@ -113,46 +114,65 @@ export function SwapPageContent() {
...
       return;
     }
+    const paprIn = paprTokenField === Field.INPUT;
+    const paprOut = paprTokenField === Field.OUTPUT;
+    const paprUniswapToken = erc20TokenToToken(paprToken, chainId);
+    const underlyingUniswapToken = erc20TokenToToken(underlying, chainId);
+
+    let baseCurrency: Token;
+    let quoteCurrency: Token;
+    if (paprIn && amounts.tradeType === TradeType.EXACT_INPUT) {
```
[the remark this answers is not in the document]
Comment (adamgobes) [reply components/SwapPageContent/SwapPageContent.tsx:162]: done

File: components/SwapPageContent/SwapPageContent.tsx (hunk 1 above)
Comment (adamgobes) [reply components/SwapPageContent/SwapPageContent.tsx:183]: added a comment on why this will never happen

**Review by wilsoncusack — DISMISSED** [review]

File: components/SwapPageContent/SwapPageContent.tsx [hunk 3]
```diff
@@ -113,30 +113,35 @@ export function SwapPageContent() {
...
           amounts.output.currency.decimals,
         ),
-        amounts.input.currency.wrapped.address,
+        underlying.id,
         amounts.output.currency.wrapped.address,
         tokenName as SupportedToken,
       );
       return quoteResult.sqrtPriceX96After;
+    } else {
+      return null;
```
Comment (wilsoncusack) [inline components/SwapPageContent/SwapPageContent.tsx:142]: do we need to clear any state here? set something to null so the predicated rate doesn't show if, e.g. they select to swap USDC to ETH?

Comment (adamgobes) [reply components/SwapPageContent/SwapPageContent.tsx:142]: we already do that in the handle token swap section

**Review by wilsoncusack — APPROVED** [review]

**Pull request merged**